weeks/week05/modules.md

```diff
@@ -1,3 +1,150 @@
 # Section 1: Modules
 
-Coming soon.
+Python modules and Python packages (which will be covered in the next section), two mechanisms that facilitate modular programming.
+
+According to modular programming concept, large tasks are broken into smaller manageable subtasks or modules. Modularized code has the following benefits:
+
+- Readability
+- Maintainability
+- Reusability
+- Scoping
+- Manageability
+
+In this section will show how to create and import a module in Python.
+
+## References:
+
+1. [Python Modules and Packages – An Introduction](https://realpython.com/python-modules-packages/#python-modules-overview)
+2. [Modules - Python Documentation](https://docs.python.org/3/tutorial/modules.html#)
+3. [The import system - Python Documentation](https://docs.python.org/3/reference/import.html)
+4. [Imports - PEP-8](https://peps.python.org/pep-0008/#imports)
+
+## 1.1 Defining Module
+
+A module is a file with the suffix `.py` containing Python definitions and statements. It is essential to know that there are different ways to define a module in Python:
+
+1. A module can be written in Python itself.
+2. A module can be written in C and loaded dynamically at run-time. A module can be written also in Java languages, when using Jython (it depends on the Interpreter, as to which one you use).
+3. A built-in module is intrinsically contained in the interpreter.
+
+The module’s name (as a string) is available as the value of the global variable `__name__`.
+
+
+## 1.2 Accessing Module's Content
+
+A module content is accessed with the `import` statement.
+
```

Comment:
Remember to add two blank lines before the headings and one blank line after before start to write or add examples. Check it along the whole content.